fhir-server/src/main/java/com/ibm/fhir/server/filter/rest/FHIRRestServletFilter.java

```diff
@@ -261,6 +279,30 @@ private HTTPReturnPreference computeReturnPref(ServletRequest request, HTTPHandl
         return returnPref;
     }
 
+    /**
+     * Checks if the FHIR version parameter in the specified HTTP header is valid.
+     * @param headerName the name of the header
+     * @param requestHeaders the headers
+     * @return the error message if FHIR version parameter in not valid, otherwise null
+     */
+    private String checkFhirVersionParameter(String headerName, Map<String, List<String>> requestHeaders) throws FHIRException {
```

Comment:
You could convert this to a void method, that  only throws exceptions if invalid, also control the message in this code, and then make it testable/self-contained. the outer catch should catch  and  mark as invalid. you could also have a special  exception for this.

Reply:
Updated to use special exception for cleaner code.